src/wallet/wallet_rpc_server.cpp

```diff
@@ -2424,12 +2434,34 @@ namespace tools
       er.message = "Failed to open wallet";
       return false;
     }
-    if (m_wallet)
-      delete m_wallet;
     m_wallet = wal.release();
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_close_wallet(const wallet_rpc::COMMAND_RPC_CLOSE_WALLET::request& req, wallet_rpc::COMMAND_RPC_CLOSE_WALLET::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_wallet) return not_open(er);
+    if (m_wallet->restricted())
```

Comment:
I don't think this method needs to be restricted.

Reply:
I didn't allow ```on_close_wallet``` in restricted to match the behaviour of ```on_stop_wallet```.

Do we want to allow a client connected to a restricted RPC server to open and close wallets at will, or not?

Reply:
Restricted is not to stop users, it's to limit potential privacy leaking information IIRC. You'll notice it's only being used on things like transfers. Not open.